packages/server/modules/workspaces/tests/integration/roles.graph.spec.ts

```diff
@@ -0,0 +1,179 @@
+import { AllScopes } from '@/modules/core/helpers/mainConstants'
+import { mapMainWorkspaceRoleToGqlRole } from '@/modules/workspaces/helpers/roles'
+import {
+  BasicTestWorkspace,
+  createTestWorkspace
+} from '@/modules/workspaces/tests/helpers/creation'
+import {
+  BasicTestUser,
+  createAuthTokenForUser,
+  createTestUser
+} from '@/test/authHelper'
+import {
+  GetWorkspaceDocument,
+  UpdateWorkspaceRoleDocument
+} from '@/test/graphql/generated/graphql'
+import {
+  createTestContext,
+  testApolloServer,
+  TestApolloServer
+} from '@/test/graphqlHelper'
+import { beforeEachContext } from '@/test/hooks'
+import { Roles } from '@speckle/shared'
+import { expect } from 'chai'
+
+describe('Workspaces Roles GQL', () => {
+  let apollo: TestApolloServer
+  let workspace: BasicTestWorkspace
+
+  const testAdminUser: BasicTestUser = {
+    id: '',
+    name: 'John Speckle',
+    email: 'john-speckle-workspace-admin@example.org',
+    role: Roles.Server.Admin
+  }
+
+  const testMemberUser: BasicTestUser = {
+    id: '',
+    name: 'James Speckle',
+    email: 'james-speckle-workspace-member@example.org',
+    role: Roles.Server.User
+  }
+
+  before(async () => {
+    await beforeEachContext()
+    await Promise.all(
+      [testAdminUser, testMemberUser].map((user) => createTestUser(user))
+    )
+    const token = await createAuthTokenForUser(testAdminUser.id, AllScopes)
+    apollo = await testApolloServer({
+      context: createTestContext({
+        auth: true,
+        userId: testAdminUser.id,
+        token,
+        role: testAdminUser.role,
+        scopes: AllScopes
+      })
+    })
+
+    const testWorkspace: BasicTestWorkspace = {
+      id: '',
+      ownerId: testAdminUser.id,
+      name: 'My Test Workspace'
+    }
+
+    await createTestWorkspace(testWorkspace, testAdminUser)
+
+    workspace = testWorkspace
+  })
+
+  describe('update workspace role', () => {
+    after(async () => {
+      await apollo.execute(UpdateWorkspaceRoleDocument, {
+        input: {
+          userId: testMemberUser.id,
+          workspaceId: workspace.id,
+          role: null
+        }
+      })
+    })
+
+    it('should create a role if none exists', async () => {
+      const res = await apollo.execute(UpdateWorkspaceRoleDocument, {
+        input: {
+          userId: testMemberUser.id,
+          workspaceId: workspace.id,
+          role: mapMainWorkspaceRoleToGqlRole(Roles.Workspace.Admin)
+        }
+      })
+
+      const { data } = await apollo.execute(GetWorkspaceDocument, {
```

Comment:
Now that `UpdateRole` also returns the workspace, you no longer need this second `GetWorkspace` operation in your test cases (the ones below too)